logic.js

```diff
@@ -302,23 +302,44 @@ function create_input(type, name, value, text)
 function apply_deck_selection(decks)
 {
     var container = document.getElementById("tableau");
-    container.innerHTML = ""; // TODO use deck.discard_deck instead
+
+    select(

```

Comment:
I realized `filter` is available as far back as IE9 so I removed the helpers entirely and just used that. I can refactor further if you think the inline version is still opaque.